garrysmod/gamemodes/terrortown/gamemode/gamemsg.lua

```diff
@@ -245,9 +249,13 @@ local function MuteTeam(ply, cmd, args)
 
    local t = tonumber(args[1])
    ply.mute_team = t
-
-   local name = (t != 0) and team.GetName(t) or "None"
-   ply:ChatPrint(name .. " muted.")
+   if t != MUTE_NONE and t != MUTE_ALL then
+   ply:ChatPrint(team.GetName(t) .. " muted.")
+   elseif t == MUTE_ALL then
+   ply:ChatPrint("All muted.")
+   else
+   ply:ChatPrint("None muted.")
+   end

```

Comment:
The team ids in ttt are:
TEAM_TERROR = 1
TEAM_SPECTATOR = 1002
TEAM_COMNECTING = 0
TEAM_UNASSIGNED = 1001

So i think 2 should work.
And yeah your code is a little bit better.
I'll add it later.